File: Tema 7/src/Trabajador.java
abstract public class Trabajador extends Persona {
private double salario;

    public Trabajador() {
    }

    public Trabajador(String nombre, String apellido, int numeroSS) {
        super(nombre, apellido, numeroSS);
    }

    public Trabajador(double salario) {
        this.salario = salario;
    }

    public Trabajador(String nombre, String apellido, int numeroSS, double salario) {
        super(nombre, apellido, numeroSS);
        this.salario = salario;
    }

    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("salario = " + salario);
    }

    public abstract void calcularSalarioMes();

    public double getSalario() {
        return salario;
    }

    public void setSalario(double salario) {
        this.salario = salario;
    }
}

File: TrabajadorHerencia/src/model/Trabajador.kt
package model

class Trabajador(nombre : String,apellido : String,dni : String,var salario: Int, var nPagas: Int)
    : Persona(nombre, apellido, dni) {
        var seguro: Boolean = false
        constructor(
            nombre: String,apellido: String,dni: String,
            telefono : Int,correo: String,salario: Int,nPagas: Int,seguro:Boolean): this(nombre, apellido, dni, salario, nPagas){
                this.seguro = seguro
                this.telefono = telefono
                this.correo = correo
            }

    override fun mostrarDatos() {
        super.mostrarDatos()
        println("Salario : $salario")
        println("Pagas : $nPagas")
        println("Seguro : $seguro")
    }

}